### PR TITLE
FIX: revamp codebase

### PR DESCRIPTION
**Pull request by ofcljaved**
#137 

- [x] Social Links: Fix the links for "Star us on GitHub" and "GitHub link" to point to 100xdev socials. Ensure all social media links are directed to 100xdev's official accounts.

- [x] NextAuth Migration: Migrate NextAuth from the beta version to the stable version.

- [x] Toaster UI: Improve the toaster UI for better user experience.
 
- [x] /jobs Page: Resolve any issues on the /jobs page.
 
- [x] Dark Theme: Ensure the dark theme is functioning correctly on all pages.

- [x] Fix the filter component
 
- [x] Fix the UI of filter component

<hr/>

## Additional Changes
 
- [x] Code revamp

### CODE REVIEW

**Comment by ofcljaved** [comment]
@VineeTagarwaL-code please review

**Comment by ofcljaved** [comment]
https://github.com/user-attachments/assets/ff802833-392b-4b2e-9491-8c2ccf5187d8

**Comment by VineeTagarwaL-code** [comment]
Good pr, the codebase is nicely structured will go over it deeply before merging

**Comment by VineeTagarwaL-code** [comment]
image
Error while trying to use the search functionality in the explore page 
image

**Comment by VineeTagarwaL-code** [comment]
Error #2 :

the search functionality seems to be messing up when any options in the workmode is checked 
image

File: src/components/job-form.tsx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+import { Button } from "./ui/button";
+import { Input } from "./ui/input";
+import { Textarea } from "./ui/textarea";
+import { Label } from "./ui/label";
+import { Switch } from "./ui/switch";
+import { useToast } from "./ui/use-toast";
+import { filters } from "@/lib/constant/jobs.constant";
+const PostJobForm = () => {
+  const { toast } = useToast();
+  const router = useRouter();
```
Comment (VineeTagarwaL-code) [inline src/components/job-form.tsx:35]: remove, not getting used anywhere

**Comment by VineeTagarwaL-code** [comment]
Code is 10/10, really liked how well structured it is, just want to make sure all bugs are fixed first and remove unused variables also with this can you add a .github workflow to test build when any pr is opened to main ?

**Comment by pawann-2000** [comment]
> Code is 10/10, really liked how well structured it is, just want to make sure all bugs are fixed first and remove unused variables also with this can you add a .github workflow to test build when any pr is opened to main ?

I'll get the workflow up by evening, any plans regarding deployment yet?

**Comment by VineeTagarwaL-code** [comment]
we can start deploying after this code is merged, will setup vercel and all

**Comment by ofcljaved** [comment]
https://github.com/user-attachments/assets/14134bc4-1c40-46c7-85de-fc732ee1820f

- [x] Filter Jobs
- [x] Unused variables and imports

Unused test was ran by eslint so it also format the document with prettier
@VineeTagarwaL-code please review

**Comment by VineeTagarwaL-code** [comment]
Lets go !

**Pull request merged**